RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/RegistrationModels.swift

```diff
@@ -101,6 +101,14 @@ struct AuthenticationParameters: Codable {
     }
 }
 
+/// The result from a registration screen's coordinator
+enum AuthenticationRegistrationStageResult {
+    /// The screen completed with the associated registration result.
+    case completed(RegistrationResult)
+    /// The user would like to cancel the registration.
+    case cancel
+}
+
 /// The result from a response of a registration flow step.
 enum RegistrationResult {
```

Comment:
The logic in this enum is becoming non-trivial so might be worth adding some tests for it

Reply:
Tests added for `FlowResult` (which I assumed is what you were referring to in this comment).